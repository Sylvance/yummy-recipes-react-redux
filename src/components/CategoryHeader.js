import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import CategoryFormInput from './CategoryFormInput'

export default class CategoryHeader extends Component {
  static propTypes = {
    addCategory: PropTypes.func.isRequired,
    userid: PropTypes.number.isRequired
  }

  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }


  handleSave = (userid, title, description) => {
    if (title.length !== 0) {
      this.props.addCategory(userid, title, description)
    }
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    return (
      <header className="header">
        <h1>Categories</h1>
        <Button color="primary" onClick={this.toggle}>Click here to add a category</Button>
        <Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
          <ModalHeader toggle={this.toggle}>Add Category</ModalHeader>
          <ModalBody>
            <CategoryFormInput onSave={this.handleSave} userid={this.props.userid}/>
          </ModalBody>
          <ModalFooter>
            <Button color="danger" onClick={this.toggle}>Cancel</Button>
          </ModalFooter>
        </Modal>
      </header>
    )
  }
}
