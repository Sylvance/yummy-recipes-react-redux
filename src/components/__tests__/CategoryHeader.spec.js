// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import CategoryHeader from '../CategoryHeader';

describe('CategoryHeader test', () => {
  it('CategoryHeader should match snapshot', () => {
    const component = renderer.create(
      <CategoryHeader addCategory={() => {}} userid={42} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
