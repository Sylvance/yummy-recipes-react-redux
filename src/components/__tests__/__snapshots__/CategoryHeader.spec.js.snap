// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryHeader test CategoryHeader should match snapshot 1`] = `
<header
  className="header"
>
  <h1>
    Categories
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Click here to add a category
  </button>
</header>
`;
